mantidproject/mantid
# Minor Frequency Domain Analysis corrections

**Description of work.**

This PR fixes the following minor issues:

* On the transform tab for Maxent the inner and outer iterations number were assigned to the other one.
* If I load a data file with lots of small time bins (e.g. HIFI 147218) it correctly insists I use a large number of data points in Maxent (values of 65536 upwards are offered). If I then switch back to other data with larger bins this limit is still enforced which makes Maxent run much more slowly than necessary.
 * For maxent change the x axis to read field in gauss rather than x axis
 * For FFT change the x axis to read frequency in Hz
 * Set the default apodization to Lorenz and the default decay to 4.4

**To test:**

Check that all of the above issues have been resolved a set of steps to do this are.
 * Open the Frequency domain analysis GUI and switch to EMU as the instrument. Load run 19489 this should be in the unit test directory.
* Go to the transform tab and select Maxent set the outer iterations to one and leave the inner iterations at 10. Click calculate and keep an eye on the results log to ensure it does the correct number of iterations. 
* Plot the resulting graph and check that the x axis has a label `Field (Gauss)`
* Switch to FFT and select `EMU19489; Pair Asym; long #1` as the workspace to fit. 
* Check that in the advanced options the default apodization is Lorenz and the default decay constant is 4.4
* Click calculate and plot the resulting workspace (All spectrum). Check that the x axis has a label `Frequency (Hz)`
* To test the second point above you will need access to the relevant HIFI data if you do not have access to the ISIS data archive contact me and I will send it to you. 
* Switch to HIFI and load run 147218, go to transform and switch to maxent in the advanced properties the Number of data points should have a minimum of 65536.
* Switch to 140000 instead and confirm that the minimum is now 4096



Fixes #25126 and #25102  


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
